server/plugin/command/create_event.go

```diff
@@ -26,121 +32,100 @@ type userError struct {
 
 func parseCreateArgs(args []string) (*remote.Event, *userError, error) {
 
-	var attendees []*remote.Attendee
-
-	attendee1 := &remote.Attendee{
-		Type: "required",
-		Status: &remote.EventResponseStatus{
-			Response: "",
-			Time:     "",
-		},
-		EmailAddress: &remote.EmailAddress{
-			Address: "joe@example.com",
-			Name:    "joe smith",
-		},
-	}
-
-	attendee2 := &remote.Attendee{
-		Type: "required",
-		Status: &remote.EventResponseStatus{
-			Response: "",
-			Time:     "",
-		},
-		EmailAddress: &remote.EmailAddress{
-			Address: "jane@example.com",
-			Name:    "jane smith",
-		},
-	}
-
-	attendees = append(attendees, attendee1)
-	attendees = append(attendees, attendee2)
-
-	// create event
-	event := &remote.Event{
-		Subject: "TestSubject",
-		// BodyPreview: "testBodyPreview",
-		BodyPreview: "DEBUG_BodyPreview",
-		Body: &remote.ItemBody{
-			Content:     "Hello!  Here is the start of Your Body!",
-			ContentType: "Text",
-		},
-		ReminderMinutesBeforeStart: 15,
-		Location: &remote.Location{
-			DisplayName:  "Las Vegas",
-			LocationType: "homeAddress",
-			Address: &remote.Address{
-				Street:          "3730 Las Vegas Blvd S",
-				City:            "Las Vegas",
-				State:           "Nevada",
-				CountryOrRegion: "US",
-				PostalCode:      "89158",
-			},
-			Coordinates: &remote.Coordinates{
-				Latitude:  47.672,
-				Longitude: -102.103,
-			},
-		},
-		Attendees: attendees,
-		Start: &remote.DateTime{
-			TimeZone: "Pacific Standard Time",
-			DateTime: time.Now().Format(time.RFC3339),
-		},
-		End: &remote.DateTime{
-			TimeZone: "Pacific Standard Time",
-			DateTime: time.Now().Add(time.Hour).Format(time.RFC3339),
-		},
-	}
-
-	// parse flags and start overriding Demo Defaults
+	event := &remote.Event{}
+
 	createFlagSet := getCreateEventFlagSet()
 	err := createFlagSet.Parse(args)
 	if err != nil {
 		return event, nil, err
 	}
 
-	//                //
-	// Required Flags //
-	//                //
+	help, err := createFlagSet.GetBool("help")
+	if help == true {
+		return nil, &userError{ErrorMessage: fmt.Sprintf(getCreateEventFlagSet().FlagUsages())}, nil
+	}
+
 	subject, err := createFlagSet.GetString("subject")
 	if err != nil {
 		return event, nil, err
 	}
 	// check that next arg is not a flag "--"
-	if subject == "" || strings.HasPrefix(subject, "--") {
+	if strings.HasPrefix(subject, "--") {
```

Comment:
??? I don't understand this check. The default value appears to be "", not "--".

Reply:
There appears to be a few limitations of the go `flag` and the `pflags` libraries.  

1. The library expects a non-bool flag to have a value and does not discriminate if the value is the next flag.  
Example `--flag --nextflag`, the value of the `--flag` is going to be `--nextflag`.  This `strings.HasPrefix(subject, "--")` checks that case and returns an error when it occurs.

However, given the example `--flag flagvalue --nextflag`, where `--nextflag` is end-of-line, the parser will let the user know there is no value for the `--nextflag` flag

2. There is know required setting for flag.  We have to define which flags are required and handle to code to ensure the flags are set.  I have this code in the following commit.

Reply:
Right, got it. 0/5 I'd have made a different choice to keep the implementation simple and clearly intention-revealing, but I understand the considerations now. 👍